wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandRobot.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj2.command;
+
+import edu.wpi.first.wpilibj.TimedRobot;
+
+/**
+ * CommandRobot is a wrapper over TimedRobot designed to work well with Command based programming.
+ *
+ * <p>The CommandRobot class is intended to be subclassed by a user creating a robot program.
+ */
+public class CommandRobot extends TimedRobot {
+  /** The CommandScheduler instance. */
+  protected CommandScheduler m_scheduler = CommandScheduler.getInstance();
+
+  /** The Command run during autonomous. */
+  protected Command m_autonomousCommand =
+      Commands.print("Default auto Command override autonomousInit to add your own.");
+
+  /** Constructor for CommandRobot. */
+  protected CommandRobot() {
+    this(kDefaultPeriod);
+  }
+
+  /**
+   * Constructor for CommandRobot.
+   *
+   * @param period Period in seconds.
+   */
+  protected CommandRobot(double period) {
+    super(period);
+  }
+
+  @Override
+  public final void robotInit() {}
+
+  @Override
+  public final void robotPeriodic() {
+    m_scheduler.run();
```

Comment:
The RobotPy version of this uses addPeriodic to run this instead. That seems like it would be a better approach since the user can't disable the scheduler on accident if they override robotPeriodic.

https://github.com/robotpy/robotpy-commands-v2/blob/main/commands2/timedcommandrobot.py#L14

Reply:
I will use addPeriodic but with the method being final they shouldn't be able to override it no?

Reply:
It can't be final in C++ or Python

Reply:
C++ can be final.